### ISSUE TITLE
Plotting pie charts at nodes base on trait reconstruction

### ISSUE THREAD

**Issue by PeteCowman** [GuangchuangYu/ggtree#31]
Hi there. Really like this package. I have being trying to figure out if the subview function can be used to annotate nodes with small pie charts based on an ancestral character reconstruction. This type of node label is possible using the ape function node label(pie = results), where results is a matrix with state probabilities at each node.

This would be a great feature to add if it is not already possible with subview!

**Comment by GuangchuangYu** [comment]
<http://guangchuangyu.github.io/ggtree/treeAnnotation-inset.html>

how about this one?

**Comment by PeteCowman** [comment]
This is excellent! I have been able to replicate all of the tutorial which is great!
I will test on some other trees and data i have from biogeographic area reconstruction!

I did notice that it does not work with layout type “fan”. when i try i get the error

Error: annotation_custom only works with Cartesian coordinates

It would be great to use the nodepie() and inset() function on a fan shaped tree. Can this be easily done?

Thanks for this function, its great! 

Pete
